app/services/PeriodHelper.scala

```diff
@@ -98,4 +110,33 @@ trait PeriodHelper {
       }
     }
 
+  def assignPayDates2(frequency: PaymentFrequency, sortedPeriods: Seq[Periods], payDay: LocalDate): Seq[PeriodWithPaymentDate] =
```

Comment:
Do you think this could go temporary in a "PeriodHelperEnhanced" or something like to keep separated - just a thought